rclpy/rclpy/parameter_client.py

```diff
@@ -0,0 +1,310 @@
+        :return: ``Future`` with the result of the request.
+        """
+        request = ListParameters.Request()
+        if prefixes:
+            request.prefixes = prefixes
+        request.depth = depth
+        future = self._list_parameter_client.call_async(request)
+        if callback:
+            future.add_done_callback(callback)
+        return future
+
+    def get_parameters(self, names: List[str], callback: Optional[Callable] = None) -> Future:
+        """
+        Get parameters given names.
+
+        :param names: List of parameter names to get.
+        :param callback: Callback function to call when the request is complete.
+        :return: ``Future`` with the result of the request.
+        """
+        request = GetParameters.Request()
+        request.names = names
+        future = self._get_parameter_client.call_async(request)
+        if callback:
+            future.add_done_callback(callback)
+        return future
+
+    def set_parameters(
+        self,
+        parameters: Sequence[Parameter],
+        callback: Union[Callable, None] = None
+    ) -> Future:
+        """
+        Set parameters given a list of parameters.
+
+        The result after the returned future is complete
+        will be of type ``rcl_interfaces.srv.SetParameters.Response``.
+
+        :param parameters: Sequence of parameters to set.
+        :param callback: Callback function to call when the request is complete.
+        :return: ``Future`` with the result of the request.
+        """
+        request = SetParameters.Request()
+        request.parameters = parameters
+        future = self._set_parameter_client.call_async(request)
+        if callback:
+            future.add_done_callback(callback)
+        return future
+
+    def describe_parameters(
+        self,
+        names: List[str],
+        callback: Optional[Callable] = None
+    ) -> Future:
+        """
+        Describe parameters given names.
+
+        The result after the returned future is complete
+        will be of type ``rcl_interfaces.srv.DescribeParameters.Response``.
+
+        :param names: List of parameter names to describe
+        :param callback: Callback function to call when the request is complete.
+        :return: ``Future`` with the result of the request.
+        """
+        request = DescribeParameters.Request()
+        request.names = names
+        future = self._describe_parameters_client.call_async(request)
+        if callback:
+            future.add_done_callback(callback)
+        return future
+
+    def get_parameter_types(
+        self,
+        names: List[str],
+        callback: Optional[Callable] = None
+    ) -> Future:
+        """
+        Get parameter types given names.
+
+        The result after the returned future is complete
+        will be of type ``rcl_interfaces.srv.GetParameterTypes.Response``.
+
+        Parameter type definitions are given in rcl_interfaces.msg.ParameterType
```

Comment:
These are [mirrored in rclpy](https://github.com/ros2/rclpy/blob/2c8ffe098eefebeec5a49e906e2506da4e0db434/rclpy/rclpy/parameter.py#L26)
```suggestion
        Parameter type definitions are given in Parameter.Type
```